contracts/Removal.sol

```diff
@@ -91,25 +114,37 @@ contract Removal is
    * @param amounts Each removal's tonnes of CO2 formatted as wei
    * @param ids The token ids to use for this batch of removals. The id itself encodes the supplier's ethereum address, a parcel identifier,
    * the vintage, country code, state code, methodology identifer, and methodology version.
+   * @param escrowScheduleStartTimes The start times, in seconds since the unix epoch, of the escrow schedules for each removal (serves as escrow schedule id in EscrowedNORI)
    * @param data Encodes the market contract address and a unique identifier for the parcel from whence these removals came.
    */
-  function mintBatch(
+  function mintRemovalBatch(
     address to,
     uint256[] memory amounts,
     uint256[] memory ids,
+    uint256[] memory escrowScheduleStartTimes,
     bytes memory data
-  ) public override {
+  ) public {
+    require(
+      amounts.length == ids.length,
+      "Removal: amounts and ids length mismatch"
+    );
+    require(
+      amounts.length == escrowScheduleStartTimes.length,
+      "Removal: amounts and escrowScheduleStartTimes.length"
+    );
+    // todo should we check that the removal id-encoded supplier address for each id is the same as `to` ?
     BatchMintRemovalsData memory decodedData = abi.decode(
       data,
       (BatchMintRemovalsData)
     );
     for (uint256 i = 0; i < ids.length; i++) {
       if (_tokenIdExists[ids[i]]) {
-        revert TokenIdExists(); // todo can the duplicate token id be reported here?
+        revert TokenIdExists({tokenId: ids[i]});
```

Comment:
Just a thought for now: I wonder how much overhead this if statement is adding and whether there's a way to to check if it exists once for all ids (e.g., via a merkle proof of some kind)

Reply:
left a gas optimization comment in the contract w this idea

Reply:
@amiecorso It's approximately 271 gas per iteration